spec/models/user_spec.rb

```diff
@@ -13,14 +13,14 @@
   #tests for change-access-code feature
   it "changes the registration code" do
     newCode = Code.regular_code + "nonsense"
-    User.change_registration_code(newCode)
+    User.change_code("regular", newCode)
```

Comment:
You started passing a "type" into your changing access code functions, but there is a string still passed in here?

Reply:
the type variable is assumed to be string, since the code_type field of the Code model is a t.string